packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -638,20 +618,75 @@ class PaddleSliderValueIndicatorShape extends SliderComponentShape {
     return _bottomLobeEnd;
   }
 
-  void _drawValueIndicator(Canvas canvas, Offset center, Paint paint, double scale,
-      TextPainter labelPainter, double textScaleFactor) {
+  // Determines the "best" offset to keep the bubble on the screen. The calling
+  // code will bound that with the available movement in the paddle shape.
+  double _getIdealOffset(
+    RenderBox parentBox,
+    double halfWidthNeeded,
+    double scale,
+    Offset center,
+    MediaQueryData mediaQueryData,
+  ) {
+    const double edgeMargin = 4.0;
+    final Rect topLobeRect = new Rect.fromLTWH(
+      -_topLobeRadius - halfWidthNeeded,
+      -_topLobeRadius - _distanceBetweenTopBottomCenters,
+      2.0 * (_topLobeRadius + halfWidthNeeded),
+      2.0 * _topLobeRadius,
+    );
+    final Offset origin = center - parentBox.localToGlobal(Offset.zero);
```

Comment:
Using localToGlobal during paint doesn't work, because you won't be notified if its value would change.
For example, consider if the slider is in a transform that is sliding left and right. You won't be repainted as it gets near the edge.

Reply:
OK, so what do I do about getting the transform then?  What I think I want is the current transform matrix for the canvas, but there doesn't appear to be a way to get that.